build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/client_constructor.rb

```diff
@@ -25,7 +25,8 @@ def initialize(options)
 
     def documented_plugin_options(plugins)
       plugins.map(&:options).flatten.select(&:documented?).sort_by do |opt|
-        [opt.required ? 'a' : 'b', opt.name]
+        # Determine sorting - required first, and add overrides at the end
+        [opt.required ? 'a' : 'b', opt.name, opt.override ? 'b' : 'a']
```

Comment:
The plugins are passed to this class in the order that they are applied to the generated client, which is the order that defaults will be resolved in as well.  We should use that ordering rather than relying on whether one of them is required or not.